lib/dial/opts.go

```diff
@@ -0,0 +1,41 @@
+package dial
+
+// Transport is a type, with a restriction on the list of supported connection modes.
+type Transport string
+
+func (t Transport) String() string {
+	return string(t)
+}
+
+const (
+	TransportDefault Transport = ""
+	TransportPlain   Transport = "plain"
+	TransportSSL     Transport = "ssl"
+	TransportInvalid Transport = "invalid"
+)
+
+func ConvertTransport(tr string) Transport {
+	switch tr {
+	case "":
```

Comment:
Or we could do something like that to avoid duplication:
```suggestion
	case TransportDefault.String():
```